### PR TITLE
xosview2: init at 2.2.2

### PR DESCRIPTION
**Pull request by SeanZicari**
###### Motivation for this change

xosview is a handy utility that I wanted to have a nix package for.

http://xosview.sourceforge.net/

This program does not yet support all of the possible libraries it can be compiled to use. From the README:

> The configure script will attempt to look for the xpm library to load
    background images with.  It will also look for the Xft library for freetype2
    font support.  If it finds the SM/ICE libraries it will configure support
    for working as a session client.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: lib/maintainers.nix
Comment (vyp) [inline lib/maintainers.nix]: Your handle here should be the same as your GitHub username.

File: pkgs/tools/X11/xosview2/default.nix
Comment (vyp) [inline pkgs/tools/X11/xosview2/default.nix]: Did you forget to fill in the description?

File: pkgs/tools/X11/xosview2/default.nix
Comment (vyp) [inline pkgs/tools/X11/xosview2/default.nix]: Use 2 space indentation for the file to be consistent with other nix expressions in nixpkgs.

File: lib/maintainers.nix
Comment (vyp) [inline lib/maintainers.nix]: I'm sorry, I didn't mean your name, I meant the `seanz` part (see the top of the file for clarification).

File: pkgs/tools/X11/xosview2/default.nix
Comment (vyp) [inline pkgs/tools/X11/xosview2/default.nix]: As mentioned at https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes, `description` should be a short one line description, so maybe just the first sentence:

    description = "Lightweight program that gathers information from your operating system and displays it in graphical form"

You could additionally put the whole thing under `longDescription` if you'd like to.

File: pkgs/tools/X11/xosview2/default.nix
Comment (vyp) [inline pkgs/tools/X11/xosview2/default.nix]: `seanz` here needs to be updated to `SeanZicari`. :wink:

**Review by vyp — APPROVED** [review]

**Comment by SeanZicari** [comment]
@vyp Is there a way to manually trigger the build? Seems like it's not starting up again.

**Comment by vyp** [comment]
@SeanZicari I'm not sure if there's a way actually. :/

**Comment by SeanZicari** [comment]
What is the next step then? How would I go about having the package validated?

**Comment by vyp** [comment]
Well in these cases you could probably just redo a commit and force push it again, so it would probably trigger a new ci build. Or in this particular case, you could also squash all those extra 'fixup' commits into one, which would probably trigger it too (and cause it'd probably be nice to squash them anyway). But of course doing these just to trigger ci builds is not ideal so I wouldn't consider them actual solutions to you question.

I should make it clear that I don't have commit access to the repo, so I can't merge this, I just wanted to help with reviewing simple issues to the PR. Also sometimes the CI builds fail due to unrelated reasons, but the packages still get accepted so I don't think it's a hard requirement to have the checks pass, but of course it is helpful to see it. FWIW to anyone reading, the two linux builds seems to pass and it looks like the mac one just seems to have been hung indefinitely somehow, and in addition travis has decided to not start any new builds, because the latest one got 'autocancelled'.

**Comment by SeanZicari** [comment]
Thanks! You’ve been very helpful.

File: pkgs/tools/X11/xosview2/default.nix
Comment (peterhoeg) [inline pkgs/tools/X11/xosview2/default.nix]: Please use the "mirror" syntax instead:

```
url = "mirror://sourceforge/xosview/${name}.tar.gz";
```

File: pkgs/tools/X11/xosview2/default.nix
Comment (peterhoeg) [inline pkgs/tools/X11/xosview2/default.nix]: Please put in a comment about why checks are disabled.

File: pkgs/tools/X11/xosview2/default.nix
Comment (peterhoeg) [inline pkgs/tools/X11/xosview2/default.nix]: If you do ```meta = with stdenv.lib; { ``` instead you can get rid of all the ```stdenv.lib.``` stuff below.

File: pkgs/tools/X11/xosview2/default.nix
Comment (peterhoeg) [inline pkgs/tools/X11/xosview2/default.nix]: Better be more specific and change ```xorg``` to ```libX11``` as well as in the ```buildInputs```. Otherwise looking very good!

**Comment by SeanZicari** [comment]
@GrahamcOfBorg What does that error mean? What do I need to resolve?

**Comment by grahamc** [comment]
hmm not sure, I'll check in to it. In the mean time, note you should add it to all-packages.nix, making it possible to use :)

**Comment by SeanZicari** [comment]
@grahamc Ah, thank you. I did have it in all-packages.nix at one point, but that must have somehow gotten lost when I was squashing commits. I'll get that done.

**Comment by SeanZicari** [comment]
@grahamc I've added the package to all-packages.nix. Just waiting for the build process to complete successfully.

File: pkgs/tools/X11/xosview2/default.nix
Comment (peterhoeg) [inline pkgs/tools/X11/xosview2/default.nix]: This is wrong. Should be without ".net":
```
url = "mirror://sourceforge/xosview/${name}.tar.gz";
```

It's good to merge after this is fixed.

**Comment by peterhoeg** [comment]
@SeanZicari thank you for your contribution and your patience!

**Comment by SeanZicari** [comment]
Thanks, @peterhoeg, but sadly I've had some trouble getting all the changes in. That last .net removal didn't end up making it in due to some bad merges. I apologize - can I fix and push the corrections?

**Comment by SeanZicari** [comment]
I was also missing myself as a package maintainer - that got left off while I was squashing some previous commits.

**Comment by peterhoeg** [comment]
Open a new PR and ping me in it.

**Comment by SeanZicari** [comment]
Will do - working on it now.

**Comment by peterhoeg** [comment]
The maintainer part is OK.

**Pull request merged**
